substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -269,6 +269,9 @@ pub mod pallet {
 		#[pallet::constant]
 		type MaxUnlockingChunks: Get<u32>;
 
+		/// The maximum amount of controller accounts that can be deprecated in one call.
+		type MaxControllersInDeprecationBatch: Get<u32>;
```

Comment:
Since its for one time migration, how about hardcoding this value in the pallet?

Reply:
If there is another runtime that is using this pallet and need to migrate, this bound might be different from them, so I guess its still better to have this as a configuration as you have done.

Reply:
Yes unfortunately the bounds are different per runtime. Generally speaking there are quite a few types that a lot of mocks don't use, could be considered more of a pallet-wide issue. The plus sides are that it is temporary, either by removing after the migration or when the staking parachain is migrated to.